flink-streaming-scala/src/main/scala/org/apache/flink/streaming/api/scala/KeyedStream.scala

```diff
@@ -79,7 +78,34 @@ class KeyedStream[T, K](javaStream: KeyedJavaStream[T, K]) extends DataStream[T]
 
     asScalaStream(javaStream.process(processFunction, implicitly[TypeInformation[R]]))
   }
-  
+
+  /**
+   * Applies the given [[KeyedProcessFunction]] on the input stream, thereby
+   * creating a transformed output stream.
+   *
+   * The function will be called for every element in the stream and can produce
+   * zero or more output. The function can also query the time and set timers. When
+   * reacting to the firing of set timers the function can emit yet more elements.
+   *
+   * The function will be called for every element in the input streams and can produce zero
+   * or more output elements. Contrary to the [[DataStream#flatMap(FlatMapFunction)]]
+   * function, this function can also query the time and set timers. When reacting to the firing
+   * of set timers the function can directly emit elements and/or register yet more timers.
+   *
+   * @param keyedProcessFunction The [[KeyedProcessFunction]] that is called for each element
+   *                             in the stream.
+   */
+  @PublicEvolving
+  def process[K, R: TypeInformation](
```

Comment:
As in java, you do not need to redefine the `K` here. So you can remove it `def process[R: TypeInformation](`...